### PR TITLE
chore: add more ip restriction tests

### PR DESCRIPTION
**Pull request by madwort**

### CODE REVIEW

File: tests/unit/jobserver/test_middleware.py [hunk 1]
```diff
@@ -85,6 +90,16 @@ def test_client_ip_middleware_proxied(rf):
     middleware(request)
     assert request.backend is None
 
+    # test spoofing header doesn't work
+    # attacker attempts to spoof both the TPP address & the proxy address
+    request = rf.get(
```
Comment (bloodearnest) [inline tests/unit/jobserver/test_middleware.py:95]: Good call, this was designed to defend against this, but didn't include the test.

For just a single proxy, the last address wouldn't be there normally

File: tests/unit/jobserver/test_middleware.py [hunk 2]
```diff
@@ -77,6 +77,11 @@ def test_client_ip_middleware_proxied(rf):
     middleware(request)
     assert request.backend == backend
 
+    # tpp client via proxy
```
Comment (bloodearnest) [inline tests/unit/jobserver/test_middleware.py:80]: This wouldn't normally happen without spoofing - the immeadiate proxy address wouldn't be in X-Forwarded-For.

So I think this case is covered in the other test.

**Review by bloodearnest — APPROVED** [review]

File: tests/unit/jobserver/test_middleware.py (hunk 1 above)
Comment (madwort) [reply tests/unit/jobserver/test_middleware.py:90]: removed the last address

File: tests/unit/jobserver/test_middleware.py (hunk 2 above)
Comment (madwort) [reply tests/unit/jobserver/test_middleware.py]: yep understand now

**Pull request merged**